src/main/java/com/iexec/core/task/TaskService.java

```diff
@@ -113,7 +113,9 @@ public TaskService(TaskRepository taskRepository,
 
         for (Task task : runningTasks) {
             String chainTaskId = task.getChainTaskId();
-            if (!replicatesService.hasWorkerAlreadyParticipated(chainTaskId, walletAddress) &&
+            boolean blockNumberAvailable = task.getInitializationBlockNumber() == 0 || task.getInitializationBlockNumber() <= blockNumber;
```

Comment:
Is it really what we want `task.getInitializationBlockNumber() == 0 ` ?

Reply:
right this should handled before.